src/main/java/javacollections/MyOwnMap.java

```diff
@@ -0,0 +1,74 @@
+package javacollections;
+
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+public class MyOwnMap implements OwnMap {
+
+    private final String[][] ownMap;
+
+    public MyOwnMap (String key, String value) {
+        this.ownMap = new String[][]{{Integer.toString(key.hashCode())}, {value}};
+    }
+
+    @Override
+    public boolean put(String key, String value) {
```

Comment:
Bardzo przekombinowana ta mapa. Powinna zostać ona zrobiona na tablicy jednowymiarowej. Używasz hashcode i equals i to jest dobre, ale Indeks to hashcode (int). 

![image](https://user-images.githubusercontent.com/17989854/127784041-9fa73d45-4fe6-4031-93a4-522d962f5059.png)